gumroad/includes/scanner.php

```diff
@@ -24,7 +24,7 @@ function gum_post_scanner( $post_object ) {
  function scanner( $post_content ) {
 
  	// counts to see if we find any gumroad product links.
- 	if (count(preg_grep('/(gumroad.com\/l\/[[:alnum:]]|gum.co\/[[:alnum:]])/', explode("\n", $post_content))) > 0) {
+ 	if (count(preg_grep('/(gumroad.com\/a\/.+\/[[:alnum:]].*|gumroad.com\/l\/[[:alnum:]]|gum.co\/[[:alnum:]])/', explode("\n", $post_content))) > 0) {
```

Comment:
It looks like in the case of `gumroad.com/l/*` and `gum.co/*` URLs, we are calling `gum_load_js()` as soon as a single `[[:alnum:]]` character is found after the last `/`. In the affiliate URL pattern, we are matching the whole URL with `[[:alnum:]].*` at the end. Wouldn't it be better to make this consistent across the pattern? Either we can drop the `.*` from affiliate URL pattern or add it to `gumroad.com` and `gum.co` URL patterns?

![image](https://user-images.githubusercontent.com/106525/116845619-ea41eb00-ac03-11eb-934a-56f13ad23462.png)

Reply:
We would like to match the affiliate product URL which for example, looks like this - https://gumroad.com/a/12345/Abc.

So, how about we have this expression?

```diff
- gumroad.com\/a\/.+\/[[:alnum:]].*
+ gumroad.com\/a\/[[:alnum:]]+\/[[:alnum:]]
```

Reply:
looks good 👍🏼

Reply:
Done - https://github.com/gumroad/wp-gumroad/pull/49/commits/a9afebd2c05220a5c68306c09cdb7ac784d22a60.